github/git_refs.go

```diff
@@ -58,13 +60,53 @@ func (s *GitService) GetRef(ctx context.Context, owner string, repo string, ref
 
 	r := new(Reference)
 	resp, err := s.client.Do(ctx, req, r)
-	if err != nil {
+	if _, ok := err.(*json.UnmarshalTypeError); ok {
+		// Multiple refs, means there wasn't an exact match.
+		return nil, resp, nil
+	} else if err != nil {
 		return nil, resp, err
 	}
 
 	return r, resp, nil
 }
 
+// GetRefs fetches a slice of Reference objects for a given Git ref.
+// If a Git ref does not match exactly, GitHub will return all refs that match it as a prefix.
+// Example:
+// feature -> [featureA, featureB]
+// featureA -> [featureA]
+//
+// GitHub API docs: https://developer.github.com/v3/git/refs/#get-a-reference
+func (s *GitService) GetRefs(ctx context.Context, owner string, repo string, ref string) ([]*Reference, *Response, error) {
+	ref = strings.TrimPrefix(ref, "refs/")
+	u := fmt.Sprintf("repos/%v/%v/git/refs/%v", owner, repo, ref)
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var rawJSON json.RawMessage
+	resp, err := s.client.Do(ctx, req, &rawJSON)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	var rs []*Reference
+	r := new(Reference)
+	// prioritize the most common case: a single returned ref
+	singleUnmarshalErr := json.Unmarshal(rawJSON, r)
+	if singleUnmarshalErr == nil {
+		return append(rs, r), resp, nil
+	}
+
+	multipleUnmarshalErr := json.Unmarshal(rawJSON, &rs)
+	if multipleUnmarshalErr == nil {
+		return rs, resp, nil
+	}
+
+	return nil, resp, fmt.Errorf("unmarshalling failed for both single and multiple responses: %s and %s ", singleUnmarshalErr, multipleUnmarshalErr)
```

Comment:
"single and multiple responses" -> "single and multiple refs". 'Responses' doesn't make sense, there's only one response.

Get rid of the trailing space here. If you don't mind, also get rid of it in `RepositoriesService.GetContents` (it's there too).